### PR TITLE
Pw 3728/3ds zeroauth

### PR DESCRIPTION
**Pull request by wboereboom**
## Summary
the ADP threedOnZeroAuth will be enforced by default on all accounts.
Currently the zeroauth flow in SFCC doesn't support 3DS

## Tested scenarios
tested using no 3ds, 3ds1 and 3ds2 test cards

### CODE REVIEW

File: src/cartridges/int_adyen_SFRA/cartridge/templates/default/account/payment/paymentForm.isml [hunk 1]
```diff
@@ -1,10 +1,20 @@
 <isset name="AdyenHelper" value="${require('*/cartridge/scripts/util/adyenHelper')}" scope="pdict"/>
 <link rel="stylesheet" type="text/css" href="${pdict.AdyenHelper.getCheckoutCSS()}"/>
+<script src="${pdict.AdyenHelper.getCheckoutUrl()}" type="text/javascript"></script>
```
Comment (maassenbas) [inline src/cartridges/int_adyen_SFRA/cartridge/templates/default/account/payment/paymentForm.isml:3]: Can you test it with the bundle.js instead of checkoutUrl?

File: src/cartridges/int_adyen_overlay/cartridge/scripts/adyenZeroAuth.js [hunk 2]
```diff
@@ -36,12 +37,19 @@ function zeroAuthPayment(customer, paymentInstrument) {
...
     zeroAuthRequest.amount = {
       currency: session.currency.currencyCode,
       value: 0,
     };
 
+    zeroAuthRequest.returnUrl = URLUtils.https('Adyen-Redirect3DS1Response').toString();
+
+    // StorepaymentMethod overrides enableReccuring/enableOneClick. Either one is allowed, not all.
     zeroAuthRequest.storePaymentMethod = true;
+    delete zeroAuthRequest.enableRecurring;
```
Comment (maassenbas) [inline src/cartridges/int_adyen_overlay/cartridge/scripts/adyenZeroAuth.js:50]: Let's delete that in the adyenHelper already where we add it in the zeroAuthRequest

File: src/cartridges/int_adyen_SFRA/cartridge/client/default/js/adyenAccount.js [hunk 3]
```diff
@@ -1,7 +1,9 @@
 const { onFieldValid, onBrand } = require('./commons/index');
 const store = require('../../../store');
 
-const cardNode = document.getElementById('card');
+let checkout;
```
Comment (maassenbas) [inline src/cartridges/int_adyen_SFRA/cartridge/client/default/js/adyenAccount.js:4]: Can't we assign the `checkout` variable already here?

File: src/cartridges/int_adyen_SFRA/cartridge/controllers/middlewares/adyen/paymentsDetails/payment.js [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+  if (
+    ['Authorised', 'Refused', 'Error', 'Cancelled'].indexOf(
+      paymentsDetailsResponse.resultCode,
+    ) !== -1
+  ) {
+    return getFinalResponse(paymentsDetailsResponse);
+  }
+
+  if (
+    [
```
Comment (maassenbas) [inline src/cartridges/int_adyen_SFRA/cartridge/controllers/middlewares/adyen/paymentsDetails/payment.js:25]: Perhaps we can extract this logic to a ResponseHandler, so we can re-use it when implementing this flow in the checkout. Let's discuss this offline

File: metadata/site_import/services.xml [hunk 5]
```diff
@@ -1,22 +1,22 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <services xmlns="http://www.demandware.com/xml/impex/services/2014-09-26">
     <service-credential service-credential-id="AdyenPayment">
-        <url>https://checkout-test.adyen.com/checkout/v65/payments</url>
+        <url>https://checkout-test.adyen.com/checkout/v66/payments</url>
```
Comment (maassenbas) [inline metadata/site_import/services.xml:4]: Was the update to v66 required for this PR?

File: src/cartridges/int_adyen_overlay/cartridge/scripts/util/adyenHelper.js [hunk 6]
```diff
@@ -52,7 +52,7 @@ var adyenHelperObj = {
   LOADING_CONTEXT_LIVE:
     'https://checkoutshopper-live.adyen.com/checkoutshopper/',
 
-  CHECKOUT_COMPONENT_VERSION: '3.18.2',
+  CHECKOUT_COMPONENT_VERSION: '3.23.0',
```
Comment (maassenbas) [inline src/cartridges/int_adyen_overlay/cartridge/scripts/util/adyenHelper.js:55]: Was this update also required? If it works with the bundle, then we can revert this change. 
Otherwise, we ideally have to update the bundle as well

**Review by maassenbas — COMMENTED** [review]
Have a few comments

File: src/cartridges/int_adyen_SFRA/cartridge/client/default/js/adyenAccount.js (hunk 3 above)
Comment (wboereboom) [reply src/cartridges/int_adyen_SFRA/cartridge/client/default/js/adyenAccount.js:4]: the declaration of the checkout variable depends on `store.checkoutConfiguration`, which implements `handleAction()` which calls `checkout`. 
If I use checkout before I declare it linting will throw errors.

File: src/cartridges/int_adyen_overlay/cartridge/scripts/adyenZeroAuth.js (hunk 2 above)
Comment (wboereboom) [reply src/cartridges/int_adyen_overlay/cartridge/scripts/adyenZeroAuth.js:50]: Discussed this over Mattermost. This wil be resolved in PW-2457

File: src/cartridges/int_adyen_SFRA/cartridge/templates/default/account/payment/paymentForm.isml [hunk 7]
```diff
@@ -1,10 +1,13 @@
 <isset name="AdyenHelper" value="${require('*/cartridge/scripts/util/adyenHelper')}" scope="pdict"/>
 <link rel="stylesheet" type="text/css" href="${pdict.AdyenHelper.getCheckoutCSS()}"/>
+<!-- <script src="${pdict.AdyenHelper.getCheckoutUrl()}" type="text/javascript"></script> -->
```
Comment (maassenbas) [inline src/cartridges/int_adyen_SFRA/cartridge/templates/default/account/payment/paymentForm.isml:3]: You can delete this line :)

**Review by maassenbas — APPROVED** [review]

**Review by maassenbas — DISMISSED** [review]

File: src/cartridges/int_adyen_SFRA/cartridge/controllers/middlewares/adyen/paymentsDetails.js [hunk 8]
```diff
@@ -0,0 +1,31 @@
+const Logger = require('dw/system/Logger');
+const URLUtils = require('dw/web/URLUtils');
+const AdyenHelper = require('*/cartridge/scripts/util/adyenHelper');
+const adyenCheckout = require('*/cartridge/scripts/adyenCheckout');
+
+/*
+ * Makes a payment details call to Adyen to confirm redirectResults and returns the resultCode
+ */
+function paymentsDetails(req, res, next) {
```
Comment (zaiddreakh) [inline src/cartridges/int_adyen_SFRA/cartridge/controllers/middlewares/adyen/paymentsDetails.js:9]: Maybe in the comments here we should explain why this is needed. Explain that there is no order but we still need to make a paymentsDetails call

**Review by zaiddreakh — APPROVED** [review]

File: src/cartridges/int_adyen_SFRA/cartridge/controllers/middlewares/adyen/paymentsDetails.js (hunk 8 above)
Comment (wboereboom) [reply src/cartridges/int_adyen_SFRA/cartridge/controllers/middlewares/adyen/paymentsDetails.js:9]: I'll elaborate on what this is used for. Note that we might use this in the future for our regular 3DS payments flow as well
